src/vmm/src/builder.rs

```diff
@@ -301,13 +300,18 @@ pub fn build_microvm_for_boot(
     #[allow(unused_mut)]
     let mut boot_cmdline = boot_config.cmdline.clone();
 
+    // Timestamp for measuring microVM boot duration.
+    let request_ts = TimestampUs::default();
+
     let (mut vmm, mut vcpus) = create_vmm_and_vcpus(
         event_manager,
         guest_memory,
         track_dirty_pages,
         vcpu_config.vcpu_count,
     )?;
 
+    attach_boot_timer_device(&mut vmm, request_ts)?;
```

Comment:
Is this method of getting the boot time generic enough to work with any other Linux guests? If not, do we want to include this device for every build of Firecracker or should we instead gate it by an optional feature?

Reply:
This is generic. The device works with all Linux guests.

The binary used in the guest is Alpine specific because our "minimal-boot" rootfs is Alpine.

Reply:
Ok, thanks.